### PR TITLE
Ravinder | Test | New Inventory

### PR DESCRIPTION
**Pull request by ravinderwalkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html [hunk 1]
```diff
@@ -71,962 +46,1197 @@
...
+                                                            id="select-option"
+                                                        ></text-field>
+                                                    </div>
+                                                    <div class="col-sm-6 col-lg-6">
+                                                        <label>Category</label>
+                                                        <div class="multi-select-input">
+                                                            <mat-chip-list #chipList aria-label="Fruit selection">
+                                                                <mat-chip
+                                                                    class="s-chip"
+                                                                    *ngFor="let categorys of category"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html:649]: Consider using plural array name
```suggestion
                                                                    *ngFor="let category of categories"
```

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.scss [hunk 2]
```diff
@@ -113,11 +162,22 @@
     width: 100%;
     padding-left: 15px;
 }
-.combo-table,
-.table {
+
+::ng-deep table.stock-table td input {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.scss:166]: `::ng-deep` is deprecated

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.ts [hunk 3]
```diff
@@ -63,11 +83,34 @@ export class CreateNewInventoryComponent implements OnInit {
...
+        const value = (event.value || "").trim();
+
+        // Add our fruit
+        if (value) {
+            this.category.push({ name: value });
+        }
+
+        // Clear the input value
+        // tslint:disable-next-line:no-non-null-assertion
+        event.chipInput!.clear();
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.ts:106]:
```suggestion
        event.chipInput?.clear();
```

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.ts [hunk 4]
```diff
@@ -63,11 +83,34 @@ export class CreateNewInventoryComponent implements OnInit {
...
+    public add(event: MatChipInputEvent): void {
+        const value = (event.value || "").trim();
+
+        // Add our fruit
+        if (value) {
+            this.category.push({ name: value });
+        }
+
+        // Clear the input value
+        // tslint:disable-next-line:no-non-null-assertion
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.ts:105]:
```suggestion
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.ts [hunk 5]
```diff
@@ -1,22 +1,154 @@
...
+
+        this.selection.select(...this.dataSource.data);
+    }
+
+    /** The label for the checkbox on the passed row */
+    checkboxLabel(row?: PeriodicElement): string {
+        if (!row) {
+            return `${this.isAllSelected() ? "deselect" : "select"} all`;
+        }
+        // @ts-ignore
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.ts:149]: Why `@ts-ignore` is used here?

File: apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.html [hunk 6]
```diff
@@ -1,86 +1,201 @@
...
+                    </mat-checkbox>
+                </th>
+                <td mat-cell *matCellDef="let row" data-title="Select">
+                    <mat-checkbox
+                        class="ml-3"
+                        color="primary"
+                        (click)="$event.stopPropagation()"
+                        (change)="$event ? selection.toggle(row) : null"
+                        [checked]="selection.isSelected(row)"
+                        [aria-label]="checkboxLabel(row)"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.html:132]: Avoid setting property with method call in template as this method will be executed on every change detection impacting the performance.

File: apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.html [hunk 7]
```diff
@@ -1,86 +1,201 @@
...
+        <table mat-table [dataSource]="dataSource" class="table giddh-table on-mobile-view">
+            <ng-container matColumnDef="select">
+                <th mat-header-cell *matHeaderCellDef>
+                    <mat-checkbox
+                        color="primary"
+                        class="ml-3"
+                        (change)="$event ? masterToggle() : null"
+                        [checked]="selection.hasValue() && isAllSelected()"
+                        [indeterminate]="selection.hasValue() && !isAllSelected()"
+                        [aria-label]="checkboxLabel()"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/new-inventory/component/inventory-product-service-list/inventory-product-service-list.component.html:121]: Avoid setting property with method call in template as this method will be executed on every change detection impacting the performance.

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**